src/ui/mainwindow.cc

```diff
@@ -304,6 +304,17 @@ MainWindow::MainWindow( Config::Class & cfg_ ):
   connect( addToFavorites, &QAction::triggered, this, &MainWindow::handleAddToFavoritesButton );
   connect( ui.actionAddToFavorites, &QAction::triggered, this, &MainWindow::addCurrentTabToFavorites );
 
+  //solo mode indicator.
+  QAction * soloAction = new QAction( QIcon( ":/icons/solo.svg" ), tr( "Quick Selection mode" ), this );
+  soloAction->setCheckable( false );
+  soloAction->setEnabled( false );
+  soloAction->setVisible( false );
+  soloAction->setToolTip(
+    tr( "Ctrl+Click a dictionary to enter the mode,Shift+Click exit the mode and restore previous selection." ) );
```

Comment:
So this tooltip give an instruction of "enter the mode", but what changed in this mode?

Reply:
users will know whether they are in the effect of this mode. which will have no such issue https://github.com/xiaoyifang/goldendict-ng/issues/2213#issuecomment-2751560299

Reply:
#2213 is a symptom, not the root problem, This fix the symptom but not the problem.

The original purpose is just temporally selecting something. Making it a mode created the problem of "checking if it is active and exit it first". If we don't make it a "mode", there will be no need for an indicator and the need of checking its status. 

The original name "solo mode" is misleading. I think it would better to call it "temporal selection" and not thinking about "mode".

Reply:
> #2213 is a symptom, not the root problem, This fix the symptom but not the problem.

any suggestion.

Reply:
I have been thinking and already suggested at least 3 other alternatives in https://github.com/xiaoyifang/goldendict-ng/issues/2213

Reply:
Will you implement this ?
IMO ,put too much time on a unfrequent used feature has little value.

Maybe I'll just accept this PR,before a new better implementation come up.

Reply:
> Will you implement this ?

Sure.

> IMO ,put too much time on an unfrequent used feature has little value.

:)